test/python/circuit/test_instructions.py

```diff
@@ -423,6 +423,267 @@ def test_repr_of_instructions(self):
+        """Test that using :meth:`.InstructionSet.c_if` with an index for the classical resource
+        resolves to the same value that :obj:`.QuantumCircuit` would resolve it to.
+
+        Regression test for gh-7246."""
+        cr1 = ClassicalRegister(3)
+        qubits = [Qubit()]
+        loose_clbits = [Clbit(), Clbit(), Clbit()]
+        # These bits are going into registers which overlap.
+        register_clbits = [Clbit(), Clbit(), Clbit()]
+        cr2 = ClassicalRegister(bits=register_clbits[:2])
+        cr3 = ClassicalRegister(bits=register_clbits[1:])
+
+        qc = QuantumCircuit(cr1, qubits, loose_clbits, cr2, cr3)
+        for index, clbit in enumerate(qc.clbits):
+            with self.subTest(index=index):
+                qc.x(0).c_if(index, 0)
+                qc.measure(0, index)
+                from_c_if = qc.data[-2][0].condition[0]
+                from_measure = qc.data[-1][2][0]
+                self.assertIs(from_c_if, from_measure)
+                # Sanity check that the bit is also the one we expected.
+                self.assertIs(from_c_if, clbit)
+
+    def test_instructionset_c_if_size_1_classical_register(self):
+        """Test that there is a distinction between conditioning on a single bit and a classical
+        register, even if the register in question has a size of one."""
+        qr = QuantumRegister(1)
+        cr = ClassicalRegister(1)
+        qc = QuantumCircuit(qr, cr)
+
+        with self.subTest("classical register"):
+            qc.x(0).c_if(cr, 0)
+            self.assertIs(qc.data[-1][0].condition[0], cr)
+        with self.subTest("classical bit by value"):
+            qc.x(0).c_if(cr[0], 0)
+            self.assertIs(qc.data[-1][0].condition[0], cr[0])
+        with self.subTest("classical bit by index"):
+            qc.x(0).c_if(0, 0)
+            self.assertIs(qc.data[-1][0].condition[0], cr[0])
+
+    def test_instructionset_c_if_no_classical_registers(self):
+        """Test that using :meth:`.InstructionSet.c_if` works if there are no classical registers
+        defined on the circuit.
+
+        Regression test for gh-7250."""
+        bits = [Qubit(), Clbit()]
+        qc = QuantumCircuit(bits)
+        with self.subTest("by value"):
+            qc.x(0).c_if(bits[1], 0)
+            self.assertIs(qc.data[-1][0].condition[0], bits[1])
+        with self.subTest("by index"):
+            qc.x(0).c_if(0, 0)
+            self.assertIs(qc.data[-1][0].condition[0], bits[1])
+
+    def test_instructionset_c_if_rejects_invalid_specifiers(self):
+        """Test that calling the :meth:`.InstructionSet.c_if` method on instructions added to a
+        circuit raises a suitable exception if an invalid specifier is passed to it."""
+
+        qreg = QuantumRegister(1)
+        creg = ClassicalRegister(2)
+
+        def case(specifier):
+            qc = QuantumCircuit(qreg, creg)
+            instruction = qc.x(0)
+            with self.assertRaises(CircuitError):
```

Comment:
Is using `assertRaisesRegex` a possibility here (just to make sure we're hitting the right `CircuitError`)?

Reply:
Oh crumbs - I was just writing the builder interface with all `assertRaisesRegex` and thinking "I'm sure I did this incorrectly only recently".  This must have been what I was thinking of!

Fixed in 9f87c4adc.